### PR TITLE
chore: release-workflow

### PR DESCRIPTION
**Pull request by d-jeffery**
## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: .github/workflows/pkg-go-release.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Language Go - Release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```
Comment (rhamzeh) [inline .github/workflows/pkg-go-release.yaml:7]: This probably shouldn't run on merge to main, just tag push

File: .github/workflows/pkg-go-build.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Language Go - Build and Test
+
+on:
+  workflow_call:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
```
Comment (rhamzeh) [inline .github/workflows/pkg-go-build.yaml:10]: Can you add fetch depth?

File: .github/workflows/pkg-go-release.yaml (hunk 1 above)
Comment (d-jeffery) [reply .github/workflows/pkg-go-release.yaml]: Artifact from before we talked about automatic releases. Removed.

File: .github/workflows/pkg-js-release.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    needs: [test]
+    permissions:
+      contents: write
+      packages: write # publish a new github release
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+
+      - uses: Roang-zero1/github-create-release-action@57eb9bdce7a964e48788b9e78b5ac766cb684803 # v3.0.1
```
Comment (rhamzeh) [inline .github/workflows/pkg-js-release.yaml:25]: Can you add the part for publishing to npm here? (see the JS SDK for an example)

File: .github/workflows/pkg-js-release.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+          path: "."
+          sparse-checkout: "pkg/js"
+          sparse-checkout-cone-mode: false
+
+      - name: Move contents of js subdirectory to CWD
+        run: |
```
Comment (rhamzeh) [inline .github/workflows/pkg-js-release.yaml:31]: Instead of this, just change cwd to pkg/js

File: .github/workflows/pkg-js-release.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      packages: write # publish a new github release
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+
+      - uses: Roang-zero1/github-create-release-action@57eb9bdce7a964e48788b9e78b5ac766cb684803 # v3.0.1
+        with:
+          version_regex: ^pkg/js/v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+
+          prerelease_regex: ^pkg/js/v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+-(alpha|beta)$
```
Comment (rhamzeh) [inline .github/workflows/pkg-js-release.yaml:63]: Can you add support for `.d+` after alpha beta?

So v0.1.0-beta.0 or v0.1.0-beta.21, not v0.1.0-beta? (would be cool to add this restriction to line 7 of this file too)

Comment (d-jeffery) [reply .github/workflows/pkg-js-release.yaml:63]: All releases or just js?

Comment (d-jeffery) [reply .github/workflows/pkg-js-release.yaml:63]: Line 7 is proving difficult as its a pattern match, not a regex.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Will something like `pkg/js/v[0-9]+.[0-9]+.[0-9]+*` work?

It should match `pkg/js/v0.2.0`, as well as `pkg/js/v0.2.0-beta`

Comment (rhamzeh) [reply .github/workflows/pkg-js-release.yaml:63]: Let's keep line 7 the same then, but abort early down the line.

Note: pkg/js/v0.2.0-beta should not be accepted - pkg/js/v0.2.0-beta.0 should be

File: pkg/java/build.gradle [hunk 6]
```diff
@@ -16,7 +16,7 @@ plugins {
     id 'io.github.gradle-nexus.publish-plugin' version '1.3.0'
 }
 
-//apply from: 'publish.gradle'
+apply from: 'publish.gradle'
 
 group = 'dev.openfga'
 version = '0.0.1'
```
Comment (rhamzeh) [inline pkg/java/build.gradle:22]:
```suggestion
version = 'v0.2.0-beta.0'
```

File: pkg/java/publish.gradle [hunk 7]
```diff
@@ -0,0 +1,60 @@
+publishing {
+    publications {
+        maven(MavenPublication) {
+            from components.java
+
+            pom {
+                group = 'dev.openfga'
+                name = 'openfga-language'
+                version = '0.0.1'
```
Comment (rhamzeh) [inline pkg/java/publish.gradle:9]:
```suggestion
                version = 'v0.2.0-beta.0'
```

File: pkg/java/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+## v0.2.0
```
Comment (rhamzeh) [inline pkg/java/CHANGELOG.md:3]:
```suggestion
## v0.2.0-beta.0
```

File: pkg/java/CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+
+### [pkg/java/v0.2.0-beta](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/java) (2024-06-10)
```
Comment (rhamzeh) [inline pkg/java/CHANGELOG.md:5]:
```suggestion
### [pkg/java/v0.2.0-beta.0](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/java) (2024-06-10)
```

File: pkg/go/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+## v0.2.0
```
Comment (rhamzeh) [inline pkg/go/CHANGELOG.md:3]:
```suggestion
## v0.2.0-beta.0
```

File: pkg/go/CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+### [pkg/go/v0.2.0-beta](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/go) (2024-06-10)
```
Comment (rhamzeh) [inline pkg/go/CHANGELOG.md:5]:
```suggestion
### [pkg/go/v0.2.0-beta.0](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/go) (2024-06-10)
```

File: pkg/go/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
...
+
+- Initial release
```
Comment (rhamzeh) [inline pkg/go/CHANGELOG.md:7]:
```suggestion
- Initial release

Note: this version does not include the validation logic that the JS and Java ports have. See issue: https://github.com/openfga/language/issues/99
```

File: pkg/js/CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v0.2.0
```
Comment (rhamzeh) [inline pkg/js/CHANGELOG.md:3]:
```suggestion
## v0.2.0-beta.18
```

File: pkg/js/CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,11 @@
...
+
+### [pkg/js/v0.2.0-beta](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/js) (2024-06-10)
```
Comment (rhamzeh) [inline pkg/js/CHANGELOG.md:5]:
```suggestion
### [pkg/js/v0.2.0-beta.18](https://github.com/openfga/language/tree/f5fc1d6b5f8585f93f87045ab41f2bb0e1a6423f/pkg/js) (2024-06-10)
```

File: pkg/js/CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,11 @@
...
+
+- Initial release
```
Comment (rhamzeh) [inline pkg/js/CHANGELOG.md:7]: Can you replace this with the changes between the two versions?

File: pkg/js/CHANGELOG.md (hunk 14 above)
Comment (rhamzeh) [reply pkg/js/CHANGELOG.md:5]: Can you also make it a compare?

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**